room/compiler/src/main/kotlin/androidx/room/solver/TypeAdapterStore.kt

```diff
@@ -532,9 +532,10 @@ class TypeAdapterStore private constructor(
                 rowAdapterLogs?.writeTo(context)
                 return rowAdapter
             }
-            if (query.runtimeQueryPlaceholder) {
+            if (query.runtimeQueryPlaceholder || query.skippedVerification) {
```

Comment:
Instead of introducing a new mutable variable we can probably just check if `resultInfo` is not null since that is an indicator that query validation was skipped. We can probably also remove the `runtimeQueryPlaceholder` and rely on the same indicator. Might be good to also update the comment on `resultInfo` to mention `@ SkipQueryVerification`.

Reply:
I agree with this sentiment and initially tried a checking if `resultInto` is null in that place. I had about 20 tests failing which made me think that there might be more cases in which `resultInfo` is not set.

Ideally I'd prefer to handle the Pojo case in line 513 where the EntityProcessor is created in case of a `@SkippedVerification` `@Entity` query. 
It might be possible to detect that a `PojoProcessor` is needed by method of elimination looking at the `XType`, i.e. if `typeElement` is not Void, not a primitive or String it's likely a Pojo.

Reply:
Where the failures from the `room-compiler` side? I think those failures would be expected since `QueryMethodProcessorTest` and some others tests expect that when verification is off, compilation fails with a 'Don't know how to convert XObject from Cursor', in other words, no adapter is found ([see code](https://github.com/androidx/androidx/blob/androidx-main/room/compiler/src/test/kotlin/androidx/room/processor/QueryMethodProcessorTest.kt#L1115)). This change updates that behaviour to now use a Pojo adapter and hope for the best if verification is off.

I think those tests would certainly need to be updated. Right now they are probably passing because `skippedVerification` is missed by them since those compiler tests directly create the processors under test and the verifier is not passed into it as if `@SkipQueryVerification` was placed in the `@Database`, but the method itself does not have the annotation. Which is actually a case not covered by the `skippedVerification` hence my request to use `resultInfo`.

Let me know if you want to try it, otherwise I can probably take over.

Reply:
You were right, I completely missed the fact that those were all pojo tests.